Register blocks

robincornett opened the pull request:
Proposed Changes:
1. Adds a build script to pull EDD Blocks into core. Use `npm run blocks` on the command line to clone.
2. Some minor tweaks to edd.css (note: still need to add the template CSS files to the minification/build).
3. Moves existing user related functions for logging in, registering, to a users directory.
4. Adds new user related functions for managing lost password requests/password resets (#3827).
5. Adds a new "success errors" option for showing success messages when calling `edd_print_errors()` (needed specifically for the forgot password flow but useful anywhere).
6. Adds a `show_required()` method to the HTML elements class to display a "required" indicator with accessible markup.
7. Adds improved no-js/js handling so that "no-js" elements do not display on screen at all (currently basically a FOUC effect due to the EDD script properly loading in the footer).
8. Updates `edd_get_purchase_link()` to output a semantically correct button instead of a link for adding items to the cart via ajax.

cklosowski commented:
@robincornett The `templates/edd.css` file is processed by `grunt build` which is part of the `npm run build` script. Do you need it to run some other time?

robincornett commented:
After doing some digging, in [7486fa4](https://github.com/awesomemotive/easy-digital-downloads/pull/9415/commits/7486fa4272a33d236d720fec72b14a491a493615) I've decided to move and handle the front end CSS files the same way the admin CSS is managed--in the assets/css directory. Because those files aren't loaded the same way as PHP template files, they don't actually need to be in the templates directory. I've updated the final fallback path for the style in `edd_register_styles` accordingly.

Noting that as part of that, I updated the two background image files to use SVG instead, so they are slightly different visually but mostly as similar as I could make them.

Currently working on creating some unit tests.
